transform/transformer.go

```diff
@@ -114,6 +189,29 @@ func (tr *Transformer) findParent(path string) (instanceTransformer, error) {
 	return parent, nil
 }
 
+// findParent walks the instanceTransformer tree to find the parent of the given path
+func (tr *TransformerXML) findParent(path string) (instanceTransformer, error) {
```

Comment:
This looks identical to the one from the JSON transformer. Unless I'm missing something looks like you could make a generic function that passes in tr.root and returns a parent.